### PR TITLE
code and .jshintrc cleanup

### PR DESCRIPTION
**Pull request by zackforbing**
I fixed a lot of small jshint errors that were showing up, and got rid of a bunch of deprecated options for jshint. 

I'm not sure this is even necessary, since we are hoping to move to ESLint, and doing so will make the PR moot.

### CODE REVIEW

File: test/header-test.js [hunk 1]
```diff
@@ -20,8 +20,7 @@ describe('<Header />', () => {
   });
 
   it('has an <Link> with LookingFor text', () => {
-    const wrapper = shallow(<Header/>);
+    const wrapper == shallow(<Header/>);
```
Comment (GSmes) [inline test/header-test.js]: @zackforbing  Should this remain a single `=`?  Originally it was setting a value for the constant `wrapper`, but changing it to `==` compares the constant `wrapper` to `shallow(<Header/>)`

**Comment by GSmes** [comment]
@zackforbing @kbs5280 @MsJennyGiraffe @Automatic365 @Laner12
I've reviewed this PR and left a single line comment in `test/header-test.js` to be reviewed before merging. Once that comment is addressed, I believe this PR can be merged.

**Comment by zackforbing** [comment]
I looked at that, not sure why that was changed. I reverted the file.

**Comment by Automatic365** [comment]
Good catch on the '='. It's definitely being set there

**Comment by Laner12** [comment]
I agree this may not be necessary for the overall goal but I think it is necessary for house cleaning. Now we can proceed on the same page.

**Pull request merged**